kedro-datasets/tests/polars/test_generic_dataset.py

```diff
@@ -0,0 +1,560 @@
+import polars as pl
+import pytest
+from adlfs import AzureBlobFileSystem
+from fsspec.implementations.http import HTTPFileSystem
+from fsspec.implementations.local import LocalFileSystem
+from gcsfs import GCSFileSystem
+from kedro.io import DataSetError, Version
+from kedro.io.core import PROTOCOL_DELIMITER, generate_timestamp
+from polars.testing import assert_frame_equal
+from s3fs import S3FileSystem
+
+from kedro_datasets.polars import GenericDataSet
+
+
+@pytest.fixture
+def filepath_csv(tmp_path):
+    return tmp_path / "test.csv"
+
+
+@pytest.fixture
+def filepath_ipc(tmp_path):
+    return tmp_path / "test.arrow"
+
+
+@pytest.fixture
+def filepath_parquet(tmp_path):
+    return tmp_path / "test.parquet"
+
+
+@pytest.fixture
+def versioned_csv_data_set(filepath_csv, load_version, save_version):
+    return GenericDataSet(
+        filepath=filepath_csv.as_posix(),
+        file_format="csv",
+        version=Version(load_version, save_version),
+        save_args={},
+    )
+
+
+@pytest.fixture
+def versioned_ipc_data_set(filepath_ipc, load_version, save_version):
+    return GenericDataSet(
+        filepath=filepath_ipc.as_posix(),
+        file_format="ipc",
+        version=Version(load_version, save_version),
+        save_args={},
+    )
+
+
+@pytest.fixture
+def versioned_parquet_data_set(filepath_parquet, load_version, save_version):
+    return GenericDataSet(
+        filepath=filepath_parquet.as_posix(),
+        file_format="parquet",
+        version=Version(load_version, save_version),
+        save_args={},
+    )
+
+
+@pytest.fixture
+def csv_data_set(filepath_csv):
+    return GenericDataSet(
+        filepath=filepath_csv.as_posix(),
+        file_format="csv",
+    )
+
+
+@pytest.fixture
+def dummy_dataframe():
+    return pl.DataFrame({"col1": [1, 2], "col2": [4, 5], "col3": [5, 6]})
+
+
+@pytest.fixture
+def filepath_excel(tmp_path):
+    return tmp_path / "test.xlsx"
+
+
+@pytest.fixture
+def parquet_data_set_ignore(dummy_dataframe: pl.DataFrame, filepath_parquet):
+    dummy_dataframe.write_parquet(filepath_parquet)
+
+    return GenericDataSet(
+        filepath=filepath_parquet.as_posix(),
+        file_format="parquet",
+        write_mode="ignore",
+        load_args={"low_memory": True},
+    )
+
+
+@pytest.fixture
+def excel_data_set_ignore(dummy_dataframe: pl.DataFrame, filepath_excel):
+    pd_df = dummy_dataframe.to_pandas()
+    pd_df.to_excel(filepath_excel, index=False)
+
+    return GenericDataSet(
+        filepath=filepath_excel.as_posix(),
+        file_format="excel",
+        write_mode="ignore",
+    )
+
+
+@pytest.fixture
+def excel_data_set_overwrite(dummy_dataframe: pl.DataFrame, filepath_excel):
+    pd_df = dummy_dataframe.to_pandas()
+    pd_df.to_excel(filepath_excel, index=False)
+
+    return GenericDataSet(
+        filepath=filepath_excel.as_posix(), file_format="excel", write_mode="overwrite"
+    )
+
+
+class TestGenericExcelDataSet:
+    def test_assert_write_mode(self):
```

Comment:
I would change this to:
```suggestion
    def test_assert_write_mode_fails(self):
```
in line with the tests below that have fail in the name for a test that should raise an exception.